server/app/repository/ApplicationEventRepository.java

```diff
@@ -0,0 +1,54 @@
+package repository;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+import com.google.common.collect.ImmutableList;
+import io.ebean.DB;
+import io.ebean.Database;
+import java.util.concurrent.CompletionStage;
+import javax.inject.Inject;
+import models.Application;
+import models.ApplicationEvent;
+
+/**
+ * ApplicationEventRepository performs operations on {@link ApplicationEvent} that often involve
+ * other EBean models or asynchronous handling.
+ */
+public class ApplicationEventRepository {
+  private final Database database;
+  private final DatabaseExecutionContext executionContext;
+
+  @Inject
+  public ApplicationEventRepository(DatabaseExecutionContext executionContext) {
+    this.database = DB.getDefault();
+    this.executionContext = checkNotNull(executionContext);
+  }
+
+  /** Insert a new {@link ApplicationEvent} record asynchronously. */
+  public CompletionStage<ApplicationEvent> insert(ApplicationEvent event) {
+    return supplyAsync(
+        () -> {
+          database.insert(event);
+          return event;
```

Comment:
Call event.refresh() after the insert completes. We do this in other repositories since EBean doesn't refresh the underlying model (i.e. setting the ID field).

Reply:
FYI @clouser2   TIL that @ Id and time annotations are updated automatically since ebean is setting them, so it's not strictly necessary here, but added it anyway for good measure.

refresh seemingly pulls in externally changed data including relationships.

Reply:
Ah, thanks! I think I'd been bitten in the past by M2M relationships not being updated or something along those lines. Thanks for adding it here. I think the consistency with the rest of the codebase will help us not to have to think about those things in the future (since database performance isn't yet an issue).